packages/app/src/components/modals/WalletInfoModal.vue

```diff
@@ -0,0 +1,160 @@
+<template>
+  <Popup :opened="opened" @close="$emit('close')">
+    <div class="wallet-info-modal-container">
+      <div class="wallet-info-modal">
+        <button @click="$emit('close')" class="close-button-icon">
+          <XIcon class="h-6 w-6" />
+        </button>
+        <span class="your-wallet-title">{{ t("walletInfoModal.yourWalletTitle") }}</span>
+        <div class="address-container">
+          <span class="address-group">
+            <HashLabel :text="formattedAddress" class="address-display" />
+            <CopyButton :value="address" class="address-copy-button" />
+          </span>
+        </div>
+
+        <div v-if="props.isWrongNetwork" class="network-mismatch-banner-ui">
+          <div class="flex items-center w-full">
+            <ExclamationCircleIcon class="h-6 w-6 mr-2 text-black" />
+            <span class="text-lg text-black flex-1"> Your wallet activated incorrect network </span>
+          </div>
+          <button class="switch-network-ui-btn" @click="switchNetwork">Switch to Prividium</button>
+        </div>
+
+        <div class="info-row">
+          <div class="balance-info">
+            <slot name="balance"></slot>
+          </div>
+          <div class="separator"></div>
+          <div class="network-info">
+            <span>{{ networkName }}</span>
+          </div>
+        </div>
+
+        <button type="button" class="disconnect-ui-btn" @click="$emit('disconnect')">
+          {{ t("walletInfoModal.disconnectButton") }}
+        </button>
+      </div>
+    </div>
+  </Popup>
+</template>
+
+<script setup lang="ts">
+import { computed } from "vue";
+import { useI18n } from "vue-i18n";
+
+import { ExclamationCircleIcon, XIcon } from "@heroicons/vue/outline";
+
+import CopyButton from "@/components/common/CopyButton.vue";
+import HashLabel from "@/components/common/HashLabel.vue";
+import Popup from "@/components/common/Popup.vue";
+
+const props = defineProps({
+  opened: {
+    type: Boolean,
+    default: false,
+  },
+  address: {
+    type: String,
+    required: true,
+  },
+  networkName: {
+    type: String,
+    required: true,
+  },
+  networkChainId: {
+    type: Number,
+    required: true,
+  },
+  isWrongNetwork: {
+    type: Boolean,
+    default: false,
+  },
+});
+
+defineEmits<{
+  (eventName: "close"): void;
+  (eventName: "disconnect"): void;
+}>();
+
+const { t } = useI18n();
+
+const switchNetwork = async () => {
+  // TODO: Implement network switch
```

Comment:
Do we have a plan for this? If we want a temporal solution maybe we can open an alert saying "comming soon" or something like that

Reply:
We have a plan, it should be integrated by today max tomorrow.